### PR TITLE
Add Ruby 2.3.0 package

### PR DESCRIPTION
**Pull request by manveru**

### CODE REVIEW

**Comment by globin** [comment]
Has the libressl patch been applied upstream?

**Comment by manveru** [comment]
Yes, I tried applying it, but it's there already.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5357,14 +5357,16 @@ let
...
+    ruby_2_3_0;
 
   # Ruby aliases
-  ruby = ruby_2_2;
+  ruby = ruby_3_0;
   ruby_1_9 = ruby_1_9_3;
   ruby_2_0 = ruby_2_0_0;
   ruby_2_1 = ruby_2_1_7;
   ruby_2_2 = ruby_2_2_3;
+  ruby_3_0 = ruby_2_3_0;
```
Comment (fpletz) [inline pkgs/top-level/all-packages.nix]: Shouldn't this be `ruby_2_3` and a few lines above for `ruby`, too? ;)

**Comment by manveru** [comment]
@fpletz oh my bad, thanks for catching that.

**Comment by manveru** [comment]
@zimbatm any chance you could try/merge this?

**Comment by fpletz** [comment]
@manveru Have you tested some ruby packages if defaulting to 2.3 introduces any regressions?

**Comment by manveru** [comment]
@fpletz well, i'm using it for about a week, but I'm not having any trouble with it, i can leave default ruby at 2_2_3 if that makes you feel better, but 2.3.0 didn't introduce any breaking changes afaict.

**Comment by fpletz** [comment]
Ok, that's what I wanted to hear. Thanks! Let's try it.

**Comment by vcunat** [comment]
Using `2.3.0` as the default causes a problem in qtruby. http://hydra.nixos.org/build/32244409/ I found a reference it's supposed to be ruby's fault: https://github.com/ryanmelt/qtbindings/issues/134

**Comment by vcunat** [comment]
I can't see anything in nixpkgs actually using it, so it's got likely a rather low priority.

**Comment by manveru** [comment]
@vcunat so we'd need to apply https://bugs.ruby-lang.org/issues/11962 ? In the meanwhile qtruby package should use a specific ruby version below 2.3.

**Comment by vcunat** [comment]
Pushed in f930b3949f0e.

**Comment by sjau** [comment]
Still failing for me with regards to qtruby.

**Pull request merged**